tests/Unit/DataStructures/Test_VectorImplTestHelper.cpp

```diff
@@ -0,0 +1,103 @@
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+SPECTRE_TEST_CASE("Unit.Utilities.VectorImplHelpers.ApplyTupleCombinations",
+                  "[Utilities][Unit]") {
+  // for more generic applications, with lambdas:
+  const int zero = 0;
+  const double one = 1.2;  // for general lambda, 1.2 casts to 1
+  const std::complex<double> two{2.5, 5.0};  // for lambda, real and casts to 2
+  std::array<std::array<bool, 3>, 3> checklist {{{{false, false, false}},
+                                                 {{false, false, false}},
+                                                 {{false, false, false}}}};
+  const auto lambda_tuple = std::make_tuple(zero, one, two);
+  TestHelpers::VectorImpl::apply_tuple_combinations<2>(
+      lambda_tuple, [&checklist](auto x, auto y) noexcept {
+        gsl::at(gsl::at(checklist, static_cast<size_t>(std::real(x))),
+                static_cast<size_t>(std::real(y))) = true;
+        CHECK(std::imag(x) + std::real(y) >= 0);
+      });
+  CHECK(checklist == std::array<std::array<bool, 3>, 3>{
+          {{{true, true, true}}, {{true, true, true}}, {{true, true, true}}}});
+}
+
+SPECTRE_TEST_CASE("Unit.DataStructures.VectorImplHelpers.TupleHelpers",
+                  "[Utilities][Unit]") {
+  auto test_tup = std::make_tuple(5.5, std::complex<int>{2, 3}, 'c');
+  CHECK(TestHelpers::VectorImpl::remove_nth<0>(test_tup) ==
+        std::make_tuple(std::complex<int>{2, 3}, 'c'));
+  CHECK(TestHelpers::VectorImpl::remove_nth<1>(test_tup) ==
+        std::make_tuple(5.5, 'c'));
+  CHECK(TestHelpers::VectorImpl::remove_nth<2>(test_tup) ==
+        std::make_tuple(5.5, std::complex<int>{2, 3}));
+  const auto addr_test_tup =
+      TestHelpers::VectorImpl::addressof(make_not_null(&test_tup));
+  CHECK(addr_test_tup == std::make_tuple(&std::get<0>(test_tup),
+                                         &std::get<1>(test_tup),
+                                         &std::get<2>(test_tup)));
+}
+
+// test the check_vectors against one another, distributions default to the cast
+// to value type of [-100,100]. The second vector type should be constructible
+// from the first vector type. For instance, calling in the order of (a double
+// type, a complex double type) will function, but not the reverse.
+template <typename VectorType1, typename VectorType2,
+          typename FundamentalValueType1 =
+              tt::get_fundamental_type_t<typename VectorType1::ElementType>,
+          typename FundamentalValueType2 =
+              tt::get_fundamental_type_t<typename VectorType2::ElementType>>
+void test_check_vectors(
+    const FundamentalValueType1& low1 = FundamentalValueType1{-100},
+    const FundamentalValueType1& high1 = FundamentalValueType1{100}) noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<FundamentalValueType1> dist1{low1, high1};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+  const auto v1 = make_with_random_values<typename VectorType1::ElementType>(
```

Comment:
In this function, the variable names could be changed to improve readability. E.g., v1 -> random_value1, s1 -> size (there is only one, so why number it), and so on.